src/common/lib/util/logger.ts

```diff
@@ -44,7 +47,9 @@ const getDefaultLoggers = (): [Function, Function] => {
   /* Can't just check for console && console.log; fails in IE <=9 */
   if (
     (typeof Window === 'undefined' && typeof WorkerGlobalScope === 'undefined') /* node */ ||
-    (global.console && global.console.log && typeof global.console.log.apply === 'function') /* sensible browsers */
+    (globalOrWindow.console &&
+      globalOrWindow.console.log &&
+      typeof globalOrWindow.console.log.apply === 'function') /* sensible browsers */
```

Comment:
Could this not be `typeof globalOrWindow.console?.log?.apply === 'function`

Reply:
Good idea, done in [8e9ffac](https://github.com/ably/ably-js/pull/993/commits/8e9ffac1cbfc09d0797d8e0ad9575536bd2ea09d)